src/status_im2/contexts/wallet/home/view.cljs

```diff
@@ -25,76 +26,110 @@
-    :balance             "€0.00"
-    :percentage-value    "€0.00"
-    :customization-color :blue
-    :type                :empty
-    :emoji               "🍑"
-    :on-press            #(rf/dispatch [:navigate-to :wallet-accounts])}
-   {:customization-color :blue
-    :on-press            #(rf/dispatch
-                           [:show-bottom-sheet {:content new-account}])
-    :type                :add-account}])
+(def add-account-placeholder
+  {:customization-color :blue
+   :on-press            #(rf/dispatch [:show-bottom-sheet {:content new-account}])
+   :type                :add-account})
 
 (def tabs-data
   [{:id :assets :label (i18n/label :t/assets) :accessibility-label :assets-tab}
    {:id :collectibles :label (i18n/label :t/collectibles) :accessibility-label :collectibles-tab}
    {:id :activity :label (i18n/label :t/activity) :accessibility-label :activity-tab}])
 
+(defn calculate-raw-balance
+  [raw-balance decimals]
+  (if (not (js/isNaN (js/parseInt raw-balance)))
+  (/ (js/parseInt raw-balance) (Math/pow 10 (js/parseInt decimals))) 0))
+
+(defn calculate-balance
+  [{:keys [address]}]
+  (let [tokens       (rf/sub [:wallet-2/tokens])
+        token        (get tokens (keyword (string/lower-case address)))
+        result       (reduce (fn [acc item]
+                               (let [total-value-per-token (reduce (fn [ac balances]
+                                                                     (+ (calculate-raw-balance (:rawBalance balances)
+                                                                                               (:decimals item))
+                                                                        ac))
+                                                                   0
+                                                                   (vals (:balancesPerChain item)))
+                                     total-values (* total-value-per-token
+                                                     (get-in item [:marketValuesPerCurrency :USD :price]))]
+                                 (+ acc total-values)))
+                             0
+                             token)] 
+    (.toFixed result 2)))
+
+(defn refactor-data
+  [accounts loading?]
+  (let [refactored-accounts (mapv (fn [account]
+                                    (assoc account
+                                           :type                :empty
+                                           :customization-color :blue
+                                           :on-press            #(rf/dispatch [:navigate-to
+                                                                               :wallet-accounts])
+                                           :loading?            loading?
+                                           :balance             (str "$" (calculate-balance account))))
+                                  accounts)]
+    (conj refactored-accounts add-account-placeholder)))
+
+(defn reagent-render
```

Comment:
hmm, I think there was already a comment about this. I don't think we should name a view `reagent-render` etc. it's more implementation details. Perhaps just call it something like `view-internal` and mark it private with `defn-`

Reply:
Done